app/controllers/application_controller.rb

```diff
@@ -24,14 +32,42 @@ def error_response(http_error, errors = {})
   private
 
   def authenticate_request
-    @current_user = Users::Authorize.call(request.headers).result
-    render error_response(:unauthorized) unless @current_user
+    @current_user, = Jwt::Authenticator.new.call(access_token_from_auth_header)
+  end
+
+  def access_token_from_auth_header
+    request.headers.fetch('Authorization', '').split(' ').last
+  end
+
+  def build_authenticate_header(error_class)
+    header_signature = [
```

Comment:
Try to get the error info from the error itself and then remove the method `header_signature_error`